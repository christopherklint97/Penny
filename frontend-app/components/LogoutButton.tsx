import React from 'react'
import { useAuth0 } from '@auth0/auth0-react'
import { FRONTEND_URL } from '../config'

const LogoutButton = () => {
  const { logout } = useAuth0()

  return (
    <button
      className="px-3 py-1 text-gray-900 bg-gray-100 rounded-md hover:bg-gray-300"
      onClick={() => logout({ returnTo: FRONTEND_URL })}
    >
      Logout
    </button>
  )
}

export default LogoutButton
