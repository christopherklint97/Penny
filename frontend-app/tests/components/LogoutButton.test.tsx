import React from 'react'
import { render } from '../testUtils'
import LogoutButton from '../../components/LogoutButton'

describe('LogoutButton component', () => {
  it('renders without crashing', function () {
    render(<LogoutButton />)
  })

  it('matches snapshot', () => {
    const { asFragment } = render(<LogoutButton />, {})
    expect(asFragment()).toMatchSnapshot()
  })
})
