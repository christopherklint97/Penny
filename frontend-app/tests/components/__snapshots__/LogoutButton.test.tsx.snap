// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoutButton component matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="bg-gray-100 hover:bg-gray-300 text-gray-900 px-3 py-1 rounded-md"
  >
    Logout
  </button>
</DocumentFragment>
`;
